### ISSUE TITLE
batch normalization

### ISSUE THREAD

**Issue by cwmeijer** [NLeSC/mcfly#42]
In the CNN model, each convolution should be followed by a batch normalization layer before the activation layer. Look up details about batch normalization in for instance the dense layers.
- [x] Model generation function updated
- [x] Write test or think of good reason why not
- [x] Manually test if anything can be trained now
  Bonus points:
- [ ] Compare performance of old CNN code with BN version. Did BN version at least improve a bit?

**Comment by dafnevk** [comment]
I tested a couple of models for both the GuinneaBissau dataset and the Pamap2 dataset.
For the GuinneaBissau, it looks like I actually got a sensable model! Well, it got 0.70 accuracy on the test set, not a whole lot, but at least it's more than ever before.
For the Pamap2 dataset, I don't have proof yet that it works better.

I didn't get any NaNs for both datasets - but this could be just a coincidence.

**Comment by vincentvanhees** [comment]
I see you implemented BatchNormalization with parameters mode=0 and axis=2. This seems sensible to me, but how sure are we about this? Would it be useful to investigate other parameter settings?
[Link](https://keras.io/layers/normalization/) to corresponding Keras documentation

**Comment by vincentvanhees** [comment]
I just ran the GuinneaBissau notebook with 50 training epochs, and left all other parameters un-touched. The results show that models overfit: the training accuracy rockets to > 0.95 for the better models while the validation accuracy never gets beyond the 0.70 Dafne found.
